services/src/main/java/org/keycloak/services/resources/admin/permissions/UserPermissions.java

```diff
@@ -262,7 +262,7 @@ public void requireManage() {
      */
     @Override
     public boolean canManage(UserModel user) {
-        return canManage() || canManageByGroup(user);
+        return canManage() || (Profile.isFeatureEnabled(Profile.Feature.AUTHORIZATION) && canManageByGroup(user));
```

Comment:
Shouldn't we check `org.keycloak.common.Profile.Feature#ADMIN_FINE_GRAINED_AUTHZ` instead?

Reply:
@pedroigor Could be possible, but I basically just copied how it is done elsewhere like here: https://github.com/keycloak/keycloak/blob/99c06d11023689875b48ef56442c90bdb744c869/services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java#L254
Its basically like that all over the place. ;)

Reply:
@pedroigor Any feedback from your side?

Reply:
Sorry. My last comment is because `AUTHORIZATION` should always be enabled. But not the `ADMIN_FINE_GRAINED_AUTHZ` feature.

So I'm wondering what difference that condition makes if you remove it or at least change it to `ADMIN_FINE_GRAINED_AUTHZ` if there is a difference.

Reply:
@pedroigor If `AUTHORIZATION`is always enabled but the `ADMIN_FINE_GRAINED_AUTHZ` is the correct one to use, it is probably used wrong all over the place. There is only one place where `ADMIN_FINE_GRAINED_AUTHZ` is used: https://github.com/keycloak/keycloak/blob/820ab52dce148eae0d51ba194ba9248bf83a69c6/services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java#L191

`AUTHORIZATION` is used much more often and I assume it's used wrong in these places:
https://github.com/keycloak/keycloak/blob/cdef39f76c50506681227b627e678bdbf15779a4/services/src/main/java/org/keycloak/services/resources/admin/permissions/UserPermissions.java#L86

https://github.com/keycloak/keycloak/blob/cdef39f76c50506681227b627e678bdbf15779a4/services/src/main/java/org/keycloak/services/resources/admin/permissions/GroupPermissions.java#L64

https://github.com/keycloak/keycloak/blob/99c06d11023689875b48ef56442c90bdb744c869/services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java#L76

https://github.com/keycloak/keycloak/blob/99c06d11023689875b48ef56442c90bdb744c869/services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java#L254

https://github.com/keycloak/keycloak/blob/99c06d11023689875b48ef56442c90bdb744c869/services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java#L265

I especially find https://github.com/keycloak/keycloak/blob/99c06d11023689875b48ef56442c90bdb744c869/services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java#L254

dangerous, as the decision that fine-grained permissions are not used is essentially taken by searching for a resource server for the realm-management client and not finding it in:

https://github.com/keycloak/keycloak/blob/99c06d11023689875b48ef56442c90bdb744c869/services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java#L259

This would mean that if fine-grained permissions were enabled before and a resource server was created for the realm management client, they would still be used even after the profile was changed to not support fine-grained permissions.

WDYT?